src/CSSTransition.js

```diff
@@ -176,8 +177,10 @@ class CSSTransition extends React.Component {
       node && node.scrollTop;
     }
 
-    this.appliedClasses[type][phase] = className
-    addClass(node, className)
+    if (className) {
+      this.appliedClasses[type][phase] = className
```

Comment:
this should be outside the conditional, no? we want to e.g. reset old values

Reply:
@taion I think that it is unnecessary to add falsy value to object and run **addClass** function, i can remove this conditional if you think that it will be better approach

Reply:
well, we want to clear any previous value that was there, no? e.g. if this already ran a transition, and the class mappings are dynamic, then we don't want to keep around an old stale value from the last time the transition ran?

Reply:
**className** will be **undefined** for this case (when **doneClassName** is **undefined**), so if it is **undefined** we could pass through: 
```javascript
this.appliedClasses[type][phase] = className
```
**appliedClasses** is used in **removeClasses** function, and it also check for **undefined**

also we can pass through:
```javascript
addClass(node, className)
```
cause it check if **className** is falsy.
So I suppose it is equivalent, but with this condition, we can skip  **addClass** and pass **undefined** value to **appliedClasses**

Reply:
i guess we wipe it in https://github.com/reactjs/react-transition-group/blob/5e2c1c4cb300b59d1c1242cb15b9607fc9caf723/src/CSSTransition.js#L193 anyway